lib/Sema/CodeSynthesis.cpp

```diff
@@ -2581,3 +2581,29 @@ swift::createDesignatedInitOverride(TypeChecker &tc,
 
   return ctor;
 }
+
+// SWIFT_ENABLE_TENSORFLOW
+ConstructorDecl *swift::getMemberwiseInitializer(NominalTypeDecl *nominal) {
+  ConstructorDecl *memberwiseInitDecl = nullptr;
+  auto ctorDecls = nominal->lookupDirect(DeclBaseName::createConstructor());
+  for (auto decl : ctorDecls) {
+    auto ctorDecl = dyn_cast<ConstructorDecl>(decl);
+    if (!ctorDecl)
+      continue;
+    // Continue if:
+    // - Constructor is not a memberwise initializer.
+    // - Constructor is implicit and takes no arguments, and nominal has no
+    //   stored properties.
+    //   - This ad-hoc logic accepts empty struct constructors generated via
+    //     `TypeChecker::defineDefaultConstructor`. Alternatively,
+    //     `defaultDefaultConstructor` could be edited to mark empty
+    //     constructors as memberwise initializers.
```

Comment:
Yes, both of those points make sense. I was hesitant to make those two changes (editing `NominalTypeDecl` and `defineDefaultConstructor`) and diverging from master.

What do you recommend? I can make those two changes, mark with `SWIFT_ENABLE_TENSORFLOW`, and upstream to master later (but without any tests).

Reply:
I think the fact that the default init will become a memberwise init rather than have a body can be observed with a -dump-ast or -emit-silgen rest. If this helps you proceed I don’t see any problem upstreamg it now

Reply:
I'm sorry, could you please be specific what changes you'd like to see? I'm a bit hazy so that'd be much appreciated!

Are you suggesting that the logic in `getMemberwiseInitializer` (introduced by this PR) be changed so that default constructors are also considered memberwise initializers? Or that `defaultDefaultConstructor` should mark all returned constructors as memberwise initializers?

Some observations:
- The derived conformances logic assumes that `getMemberwiseInitializer` returns a constructor whose parameter count is equal to the number of stored properties in the nominal type. Marking default constructors (which take zero parameters) changes this contract and would complicate caller code.
- There doesn't exist a flag like `isMemberwiseInitializer()` for default initializers. We could check other conditions like `isSynthesized()`, but those aren't necessarily accurate (other initializers could be synthesized).